### DISCUSSION TITLE
Problems setting up shared data "other server"

### DISCUSSION THREAD

**Discussion by Sagnasty** [rern/rAudio#1057] [Fixed Issues]
Was successfully using shared data (NFS NAS) for quite some time prior to updating to rAudio 20221007- extremely useful feature when trying to maintain multiple systems on both local and remote networks (DDNS).

The system in question was recently reflashed from image 20220904.

Today, I updated my system from 20220917 to rAudio 20221007 and have the following observations with configuring the new version of shared data:

Setting up shared data "other servers":
1. To prepare for the shared data update, I removed all shared data files and folders from my shared data folder which were created in previous releases and stored them in a separate location.
2. I updated a working system from 20220917 to rAudio 20221007.
3. I then applied the update branch release (prior to this, I had similar problems as reported by @ceejayemm).   The Update branch appears to have alleviated problems that made the system completely unusable.
4. I clicked on the System / Shared data client slider and proceeded to configure a shared data NFS server using my local NAS IP address and shared data folder that I had been using for shared data in previous releases.   I noticed that the slider bar returned to the "OFF" state after a few seconds (I was not expecting that).    
5. The system seem to correctly create a new mount to the shared data folder, but no shared data files were saved to the shared data folder as in previous releases.   

Not sure if there are more steps to configuring the shared data server (new shared data config instructions are a bit confusing to follow).

Some suggestions going forward:
I think it would be much clearer if a separate control existed for shared data **client** and **server** setup in the system page /  settings and data area.   The server setup instructions was a bit convoluted since it was located within the client setup (and the first machine shared date mount would then be the data used if I interpreted the instructions correctly).
Perhaps moving the new rAudio server setup from the features page and into the System server control (to keep all related functions under the systems page).
I found the shared data slider control going back to the off position a bit strange.   If the slider control position were maintained, then the user would have a better idea when the feature is active (and not active).
Include a clear indication of how the machine is configured (shared data server or client).  

Let me know if I can provide any additional data - this is a super useful feature.

Jim

3 x Raspberry Pi 3B+ Rev 1.3 > HiFiBerry DAC+ Pro
1 x Raspberry Pi 3B Rev 1.2 > HiFiBerry DAC+ Pro
Synology DS718+

**Comment by ceejayemm** [comment]
I have to agree with @Sagnasty here.  It would be far simpler and clearer to designate one installation as the 'Server' and all the rest as 'Clients' . In my case the server would be my RPi 3B+ (Office) system which has a wired Lan connection with the rest being wireless clients.  I can't yet comment on the placing of the various items on pages because I can't yet get that far but the comments above again seem reasonable - although perhaps using the Networks page rather the System page as there is currently a lot of free space there and the subject is network rather than system related ?

@Sagnasty - how did you manage to apply the Update branch at point 3 above without access to the 'Cog' menu ?

Regards.

Chris

**Comment by rern** [comment]
Hi Jim and Chris,

Please try:
- Disable `Shared Data`
- Update to UPDATE branch
- Re-enable `Shared Data` again

**Reply by Sagnasty** [reply]
Hello Rern and Chris,

In an attempt to disable shared data, I unmounted / forgot the shared data definition which immediately resulted in an automatic library refresh with no action from me.

For the sake of time, I then used another memory card which was already flashed and running with 20220917 (shared data disabled).

I updated the system directly from the update branch, configured the NFS shared data "other server" with the same info I used yesterday.  The system created the shared data mount and then populated the info into my NFS Shared data folder on my NAS and kept the slider bar in the ON state - **Bravo**!
The only issue I saw was that the WebRadio station art displayed on the library page was the default rAudio station art instead of what I created in previous releases (Station art files was re-created on the Shared data drive however).   For a quick test, I successfully changed the station art for one of the webradio stations and it was successfully updated on the shared data folder.

@ceejayemm - I think I was able to see the cog window only because I had shared data disabled before I updated (my main system - Raspberry Pi 3B+ Rev 1.3).  I tried updating my local client system (Raspberry Pi 3B Rev 1.2) which had shared data enabled, and this resulted in no GUI display (I reflashed this system back to 20220917).   I also try to keep my systems as straight forward as possible.

Jim

3 x Raspberry Pi 3B+ Rev 1.3 > HiFiBerry DAC+ Pro
1 x Raspberry Pi 3B Rev 1.2 > HiFiBerry DAC+ Pro
Synology DS718+

**Comment by ceejayemm** [comment]
@rern 

I have applied the latest Update branch and now seem to be having some success. Having applied the Update this is what I did:

Before applying the Upgrade branch I  DISABLED ALL EXISTING SHARED DATA SETTINGS on all my systems.

Upgraded to UPDATE branch (10 Oct 2022 or later) on all required systems

On Server system:
- Made sure my Library and other settings were as I wanted them, including a complete rebuild of the database (probably not necessary with hindsight).
- Features > Server rAudio > Enable
- (Do NOT enable System > Settings and Data > Shared Data)

On Client systems:
- System > Settings and Data > Shared Data > Enable
- Choose 'rAudio' and enter IP Number of rAudio Server above

Notes on use:
- using CIFS/NFS options allows shared data to be stored on separate non rAudio system as previously
- on Client systems, Two new NFS shares will be added to System > Storage, these are the shared data from the Server system :
		/mnt/MPD/NAS/SD
		/mnt/MPD/NAS/Data
- Playback screen display options are set by Server (why?) but Volume, artwork etc are individual to each Client
- When Shared Data is disabled on the Client then original local Music share is NOT restored but all other local settings are.
- Client access to Server does not seemed to be dropped (on the Server) when a Client is disabled until the Server is rebooted.

@Sagnasty - I was able to apply the Update branch (this morning) using:

curl -sfLO https://github.com/rern/rAudio-1/raw/main/install.sh
chmod 755 install.sh
./install.sh "r1 Update UPDATE"

I think you are using SharedData slightly differently to me in that you are storing the SharedData on a separate, non-rAudio-1, system (ie your NAS) which is what I have been doing up until the latest release implemented an NFS server within rAudio-1.  My observations above  are now based on the latter setup which now appears to work albeit with the observations above.

@rern  - Thanks for your efforts with thsi I think we are getting there.

Chris

**Reply by Sagnasty** [reply]
@ceejayemm -  Yes, I am using my NAS as the NFS server for shared data as I require DDNS (two additional systems on a remote network).

I did notice that my Album count was gone after I updated my library (only changed files).   I updated the library (update entire database) and this solved the issue and correctly updated the shared data.  

After rebuilding the library database, I noticed that the response time when clicking the various items (web radios, album, album artist, etc.) in the library page **was significantly improved** (almost instantaneous).   Very nice @Rern !!

Jim

3 x Raspberry Pi 3B+ Rev 1.3 > HiFiBerry DAC+ Pro
1 x Raspberry Pi 3B Rev 1.2 > HiFiBerry DAC+ Pro
Synology DS718+
